modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/upgrade/OnboardingUpgradeScreen.kt

```diff
@@ -253,19 +253,30 @@ private fun UpgradeContent(
     pages: List<UpgradePagerContent>,
     modifier: Modifier = Modifier,
 ) {
-    Column(modifier = modifier) {
-        val pagerState = rememberPagerState(initialPage = 0) { pages.size }
-        val coroutineScope = rememberCoroutineScope()
-        VerticalPager(state = pagerState, modifier = Modifier.fillMaxSize()) { page ->
-            when (val currentPage = pages[page]) {
+    val coroutineScope = rememberCoroutineScope()
+    val listState = rememberLazyListState()
+    FadedLazyColumn(
+        modifier = modifier,
+        state = listState,
+    ) {
+        itemsIndexed(pages) { index, page ->
+            when (page) {
                 is UpgradePagerContent.Features -> FeaturesContent(
-                    features = currentPage,
-                    onCtaClick = { coroutineScope.launch { pagerState.animateScrollToPage(pages.size - page) } },
+                    features = page,
+                    onCtaClick = {
+                        coroutineScope.launch {
+                            listState.scrollToItem(pages.size - index)
+                        }
+                    },
                 )
 
                 is UpgradePagerContent.TrialSchedule -> ScheduleContent(
-                    trialSchedule = currentPage,
-                    onCtaClick = { coroutineScope.launch { pagerState.animateScrollToPage(pages.size - page) } },
+                    trialSchedule = page,
+                    onCtaClick = {
+                        coroutineScope.launch {
+                            listState.scrollToItem(pages.size - index)
```

Comment:
This logic isn't quite right. If there are 2 pages, the results will be:

* Page 0: `2 - 0` → `2`. There is no page 2, but it will scroll to the last item because `scrollToItem` has graceful logic.
* Page 1: `2 - 1` → `1`. It doesn't scrolls anywhere because the page scrolls to itself.

We need something like `val nextIndex = (index + 1) % pages.size`.